Add Halon production to electrolyzer reactions, remove normal one

Ghilker opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Remove halon reaction from gas reactions.
Added it to the electrolyzer reactions slightly modified to become a tier 1 gas, now is made with co2 and n2o at low temperature and low pressure (similar to BZ)
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Allow earlier and easier uses of this gas that is supposed to be helping stop fires
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
remove: removed halon from normal gas reactions
expansion: added halon to the electrolyzer reactions by consuming co2 and n2o at low temperature and low pressure
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Ghilker committed:
halgone

JohnFulpWillard commented on code/modules/atmospherics/machinery/components/electrolyzer/electrolyzer_reactions.dm:
```diff
@@ -60,4 +60,27 @@ GLOBAL_LIST_INIT(electrolyzer_reactions, electrolyzer_reactions_list())
 	var/proportion = min(air_mixture.gases[/datum/gas/hypernoblium][MOLES], (1.5 * (working_power ** 2)))
 	air_mixture.gases[/datum/gas/hypernoblium][MOLES] -= proportion
-	air_mixture.gases[/datum/gas/antinoblium][MOLES] += proportion * 0.5
+	air_mixture.gases[/datum/gas/antinoblium][MOLES] += proportion * 0.05
```
was this intended?

Ghilker replied:
Ops was thinking of changing this one to lower it a bit (is antinob, should not be this easy, and I was supposed to put it at 0.01) will revert it

Ghilker commented on code/modules/atmospherics/machinery/components/electrolyzer/electrolyzer_reactions.dm:
```diff
@@ -60,4 +60,27 @@ GLOBAL_LIST_INIT(electrolyzer_reactions, electrolyzer_reactions_list())
 	var/proportion = min(air_mixture.gases[/datum/gas/hypernoblium][MOLES], (1.5 * (working_power ** 2)))
 	air_mixture.gases[/datum/gas/hypernoblium][MOLES] -= proportion
-	air_mixture.gases[/datum/gas/antinoblium][MOLES] += proportion * 0.5
+	air_mixture.gases[/datum/gas/antinoblium][MOLES] += proportion * 0.05
```
```suggestion
	air_mixture.gases[/datum/gas/antinoblium][MOLES] += proportion * 0.5
```

Ghilker committed:
Update code/modules/atmospherics/machinery/components/electrolyzer/electrolyzer_reactions.dm

Cheshify commented:
Incredibly based and good PR Ghilker, this is perfectly in line with what I think atmospherics needs

Kylerace commented:
how is this an expansion

MassacreDemigirl committed:
Makes Cyborg Lollipops Consistent (#65621)

tgstation-server committed:
Automatic changelog generation for PR #65621 [ci skip]

Capsandi committed:
[READY] Remap of Crashed Ship ruin (#65672)

tgstation-server committed:
Automatic changelog generation for PR #65672 [ci skip]

Incoming5643 committed:
Adds a very rare sussy firelock (#65677)

On a VERY low chance (1/25000) a normal full tile firelock can instead be generated with a more suspicious appearance based on this recent quality shitpost.

Sussy firelocks are completely identical when open (save the desc) and thus are apt not to be noticed except during exciting times. It's a fully cosmetic joke that got me to log into github, and that's an accomplishment in itself. Probability rate fully negotiable, I've added a one in a million chance before after all.

AnturK committed:
Fixes pathfinding not checking diagonal passability properly. (#65665)

In short, it was checking diagonal passability directly instead of every partial-step.
Now this obviously makes this pricier.
Partially alleviates the secbot pathing doomstack stacks since they will now properly fail to path but i still need to find the breakpoint where the image churn causes the actual client crash.
Also removes redundant path calculation in bot code that would be overwritten on step anyway.

AnturK committed:
Fish refactor. (#65785)



Refactors fish into proper paths.
Removes aquarium behaviour intermediary datums.
Moves fish functionality out of aquarium content component.
Fixes flopping animation resetting on dropping.

Simplifies everything. There should be no player facing changes.

The animate signal is kinda weak method of solving the animate queue issue but it seemed least intrusive to me.
Open to any better ideas.

actions-user committed:
Automatic changelog compile [ci skip]

Pickle-Coding commented:
The heat capacity in this reaction will increase, putting a limit on where the temperature will converge. The limit is almost below TCMB due to the low energy release from the reaction. The reaction is more efficient at lower pressures (decreasing temperature will increase reaction efficiency).

To keep it interesting, I suggest increasing the energy release so it doesn't converge to such low temperatures, and instead fight against its optimal production rate.

LemonInTheDark committed:
[MDB IGNORE] TGUI library, plus polish (#65326)

* Prevents potentially infinite length books from being written and stored. I'm not sure if this is an actual issue, but I have a funny feeling it may become one someday

* Moves the paper defines to their own file

* It's become clear to me that I am stupid

* git add --all

* Makes book info into a datum to allow for easy passing around

* Converts the library scanner to tgui, lays the groundwork for tgui visitor consoles

* Makes the db request for book info sort
Adds the frontend for the visitor's console
Adds a hash to prevent duplicate db requests
Adds a prams changed var to help facilitate a better search button
Makes the page number code accept text as input

* Makes the ui index at 1 even tho we index at 0 internally

* Begins the conversion process for the library console.

Changes the library console to override the visitor, to utalize for the archive access portion of the ui
Makes scanner into a weakref, I'm coming for you handheld scanner
Renames some vars to make things clearer

* Converts the remaining refs of the old console typepath over, adds a circuit board for consoles because pain

* Changes how bookshelves load in books
Instead of loading them in lazyally, we load them during init
This lets us track what books are stored in which areas

Somewhat jutting off of this, adds map config for designating something as "part of the library"
This will be useful later

* Renames the random poster, adds a spritesheet for bibles. Both will be useful in a moment

* Ok. This is a bit of a mess.
Converts the library console to tgui.
This comes with a few minor behavior changes:
You can now select what type of poster you want to print, instead of just printing a random one
It's now possible to heed the console's emag warning
The console's inventory page will fill at roundstart with the books in your area/if you're in a library, any
areas designated as "library like" in the map config
You can see what type of bible the chaplin has selected?

"Fixes":
You can no longer just dump books into the scanner forever

Implementation details:
Any input that makes a db request will now A: freeze up any other db inputs until it's finished, and B: Start a
1 second timer before any new db requests can be made
Of note, I'm handling html encoding in a very targeted way.
All book_data datums need to have html encoded values. get_title/author/content exist so a defaulting and tgui
appropriate version can be loaded in. This somewhat matches with the trusted var on set_title, it exists to
prevent double html encoding.

While we're here

Input/DB (Book data should be html encoded)
Inside book datum (Book data should be html encoded)
Sending to tgui (Book data should be decoded during extraction with the get_() procs)
Sending anywhere else (Book data should be html encoded, otherwise it's an xss vuln)

Uhhhh tgui stuff?
I'm using a custom theme for emag visuals, I'll get into that more later
The visitor and book management console share the same data/act pipeline, which is why they're parented/subtyped
They also share a page selection component, which is why the visitor's console imports it.

Uhhhhh
Oh right, fuck.
Ok so the page selection component is kinda cursed, the left and right controls are fine
But I'm trying to get a << < [page/max] > >> setup going, and that means resetting the center input past change
so the default value can be used
This ends up being slightly hacky. I'm sorry.

Oh also, I implemented a custom tab setup for this ui. I have no idea why it was literally like 5 months ago.
I think it looks pretty nice, but if you want me to nuke it I can. Sorry for any headache around this.

More tgui stuff next

* Scanner/visitor cleanup, some other odds and ends

* Adds in a dark red and black theme for library computers to be triggered by an emag.
Things of note: I'm overriding some lists that get passed into buttons and one other thing using set, since the
list is alreadt generated by that step in the process? I think?
I've added dimness control to the dimmer component, since well, it was dimming already dark uis.
I also made and added a rather large background svg. I've got no experience with this sort of thing, and all the
compression methods I found for this ended up being busts. I know this isn't acceptable as an end product, but I
don't know how to get it there.

Somewhat on that note, this ui might not be worth the size for the amount of use it gets. I'm fine with nuking
it if that's the case, I bring this up because I have a very poor understanding of the logistics of something
like this, so I have a feeling I've fucked up somewhere

* Forgot these, just a scss file for library computers, barely used but I think it's worthwhile

* Missed this eariler. As a part of the uploading tab, I'm displaying the contents of books. I'm loading in that
context as raw html so paper -> book books look close to right. Means I need more html tags then our current
sanitize provides. I don't think any of these will cause issues, and there's also a good chance I'm missing
some. Will come up with a list later

* Updates the rest of the maps to use the new management typepath

* Fixes the default bible name being Default Bible Name, I am sorry

* Turns out I had the scaling wrong for bible names, lead to weird stacking because the bible icon doesn't scale, so I lowered its sizing

* Yeets unneeded exports (Thank you jlsnow)

Haha wouldn't it be funny if I didn't know how components worked

Co-authored-by: Jeremiah <42397676+jlsnow301@users.noreply.github.com>

* Resets the maps to master

* Fixes oversights from merge commit, changes maps

* Removes needless Flex's from the scanner

* Gives the library console the ability to parse markdown. Expands the list of acceptable html elements a bit

* Adds audio cues for printing and inserting/removing from the scanner, makes the scanner nicer to use in general

* Uses a compressed version of the background. It's still huge, but smaller at least

* Adds the printing audio to the book binder

* Cleans up tram

* curse you tram

* AHHHHHHHH

* MY LIFE IS TRUE PAIN

* Adds a path conversion statement to make people's lives easier

* Apply's style's suggestions

thx style

Co-authored-by: Aleksej Komarov <stylemistake@gmail.com>

* Compresses the background svg

* Further js cleanup

* We no longer render markdown in the ui, since any source of markdown is converted to html anyway

* More ui changes

Makes the tab/main screen logic use Flex rather then manuel offsets
Makes modals better fit the size of their contents
Readjusts the width of some inputs
Properly uses the header prop for a table
Makes the buttons in the upload panel look nicer
Restructures the print tab a bit

* Increase a modal's size

* Fixes computers with no keyboard overlay showing their screen even when the power is out

* Moves some data and logic onto the library subsystem. Kyler's review

Fixes harddels held by the library scanner. Makes the scanner's buffer
actually do something

* Makes book icon randomization a proc rather then just copypasta'd code

* Removes the kilo library edit, the soul was removed

* Damn you san (Fixes mapconflicts)

* Pain

Co-authored-by: Jeremiah <42397676+jlsnow301@users.noreply.github.com>
Co-authored-by: Aleksej Komarov <stylemistake@gmail.com>

tgstation-server committed:
Automatic changelog generation for PR #65326 [ci skip]

dragomagol committed:
add GBP: No Update label when title includes [no gbp] (#65739)

Co-authored-by: tattle <article.disaster@gmail.com>

Imaginos16 committed:
The CentCom-ening Part 4: How I lost my Marbles in the Green (#65794)

imageadd: The CentCom Officer's Coat!

tgstation-server committed:
Automatic changelog generation for PR #65794 [ci skip]

Ghilker committed:
thermomachine upgrades (#65470)

You can now upgrade the thermomachines by using 3 sheets of metallic h2 on them.
This upgrade allows them to consume all the heat without needing to connect the waste port to any gas source.

Why It's Good For The Game

Upgradeable thermomachines and more uses for the metallic hydrogen

thanks to the latest discoveries, metallic hydrogen can be installed inside the thermomachines to allow their use without needing a thermal port connected to a gas reservoir (a total of 3 metallic h2 sheets are required, obtain them from the local crystallizer, warranty void if applied to other machines)

tgstation-server committed:
Automatic changelog generation for PR #65470 [ci skip]

TheBoondock committed:
Fixes conveyor runtime (#65788)

Conveyor would runtime whenever it is right clicked with an item

Fixes #64595 (Runtime on conveyor for right clicking)

fixes a runtime with conveyor where right clicking it with any item would cause a runtime

Mothblocks rant from the issue report below, you've been warned:

Because right-clicking in BYOND is horse-shit. It pipes it all through the normal Click and only tells you it's a right-click through a flag. This means that on anything that isn't prepared, right-clicking is the same as left-clicking, which is terrible UX that only exists in SS13.

Nothing should return ..() from attackby_secondary, because the default is the legacy behavior of making right-click pass as left-click (which I want to kill ASAP, once nothing uses the stupid flags anymore).

Remove else return ..(), and make this whole thing do return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN.

tgstation-server committed:
Automatic changelog generation for PR #65788 [ci skip]

MassacreDemigirl committed:
Adds missing descriptions to some open/misc turfs (#65782)

tgstation-server committed:
Automatic changelog generation for PR #65782 [ci skip]

MrMelbert committed:
Better tomato logging (#65793)

tgstation-server committed:
Automatic changelog generation for PR #65793 [ci skip]

TemporalOroboros committed:
Makes the beaker panel send assets to the user instead of null. (#65756)

GoblinBackwards committed:
Fixes slime core removal surgery (#65754)

tgstation-server committed:
Automatic changelog generation for PR #65754 [ci skip]

Ghommie committed:
Extra-large painting canvases: 36x24 and 45x27 (#65642)

tgstation-server committed:
Automatic changelog generation for PR #65642 [ci skip]

cacogen committed:
Pull tail screentip works properly (#65729)

tgstation-server committed:
Automatic changelog generation for PR #65729 [ci skip]

Cenrus committed:
Removes stray pixel in the CQC manual sprite (#65787)

tgstation-server committed:
Automatic changelog generation for PR #65787 [ci skip]

Fikou committed:
protector stands no longer remove all their overlays when leaving shield mode (#65765)

fixes #50168

tgstation-server committed:
Automatic changelog generation for PR #65765 [ci skip]

TiviPlus committed:
delete fivestar sprite (#65812)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

san7890 committed:
Trimming The Newscaster Fat - A Lot Of Mapping [MDB IGNORE] (#65799)


About The Pull Request

Hey there,

#65038 was quite good for newscasters, but since it merged bounty boards and newscasters in the same /obj/ and replaced all instances of either, you would end up with stuff like this:

Yes, those are all on the same direction. So, I manually reviewed every single newscaster appearance across all of our maps that have newscasters and ensured that the placement/duplication was sane. I also noticed that some maps had built stuff to account stuff for the old shape of the old newscasters, so that was rearranged as seen.
Why It's Good For The Game

image

Newscaster (singular) is good. Newscasters (plural) is iffy at best.

Also, some of the new newscasters were buried under posters. Who knows how long they've been under there. I've updated them to what I felt would make the most sense from a mapping standpoint.
Changelog

cl
fix: Nanotrasen realized how much they were spending on newscasters that got buried under posters and the like, so the corporation decided to stock the crew with juuuuust enough. They have saved a lot of money from this venture, and the economy will promptly begin to skyrocket to new heights.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65799 [ci skip]

TiviPlus committed:
Minor mech_equipment/proc/detach code improvements (#65811)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

TheBonded committed:
Update mulebot.dm (#65744)

Fixed mulebot movement animation to be smoother

tgstation-server committed:
Automatic changelog generation for PR #65744 [ci skip]

cacogen committed:
Stops machine frames and circuitboards referring to manipulators as micro-manipulators (#64878)


About The Pull Request

    Gives stock parts a base_name variable and sets it to manipulator for micro-manipulators
    Reorders sprites of stock parts in stock_parts.dmi to be together and in order of tier. Image
    Makes component part lists on machine frames and circuitboards nicer and makes them identical between the two
    Improves code for generating these lists
    Adds numtotext() proc which these lists use

Why It's Good For The Game

    Solves the issue where examining machines says they require a micro-manipulator when any manipulator will do
    Makes sprites easier to find
    Wows players by using actual words for numbers in generated parts lists
    Cleans up ancient code
    This proc can be used for other things, e.g. station names (which I have now incorporated it into)

Changelog

cl
fix: Machine frames and circuit boards no longer call manipulators micro-manipulators when examined
code: Improves code in constructable_frame.dm and circuitboard.dm
/cl

tgstation-server committed:
Automatic changelog generation for PR #64878 [ci skip]

LemonInTheDark committed:
Action button refactor/rework: Enhanced Dragging (#65180)

About The Pull Request

I noticed a lot of strange and un-intuitive behavior in action buttons, and got stung by the bloat bug. Damn it hug #58027
I'll do my best to explain what I've changed and why, might get a bit long.
If you want a better idea, read the commits. Most of em are pretty solid, if long.

Whelp. Here we go.
How do action buttons currently work

All action buttons are draggable, to any place on the screen. They're held in an actions list on the player's mob.
Their location in this list determines their position on the top of the screen. If one is dragged away from the top, its position in the list is "saved". This looks really bad.
If two buttons are dragged over each other, their positions swap. (inside the actions list too)
If a button is shift clicked, it is brought back to the position it started at.
If the action collapse button that you likely just mentally edit out is alt clicked, it resets the position of all action buttons on the screen.
If an action is ctrl clicked, it is "locked". This prevents any future position changes, and also enables a saving feature. With this saving feature, locked button positions persist between rounds. So your first o2 canister will always start where you saved it, etc.
Actions and buttons are a one to one link. While there is functionality to share action buttons between two players, this means showing the same object to both. So one player can move a button on another's screen. Horrendous.
This also makes code that modifies properties of the screen object itself very clunky.
Why is this bad

A: None knew pretty much any of this information. It is actually documented, just in a horribly formatted screen tip on the collapse button, you know the one we all mentally delete from the hud.
B: None of this is intuitive. Dragging buttons makes the hud look much worse, and you get no feedback that you even can drag them. Depressing
C: We use actions to make new options clear to the player. This means players can have a lot of action buttons on the hud. This gets cluttery
D: The collapse button is useless. It lets you clear your screen if someone like me fucks up and gives you 2000 actions, but outside of that it just hides all information from you. You never want to see none of your action buttons, just a filtered list of them.
E: On a technical level, they're quite messy, and not fully functionally complete. This is depressing.
What I've done

Assuming the above to be true, how do we fix them?
Well first I'm going to go over everything I changed, including links to major commits. I'll then describe the finished product, and why I made the decisions I did.

Oh and I've moved some of the more niche or technical discussion to dropdowns. Hopefully this makes finding the major functional changes easier

Adds helper procs for turning screen_loc strings into more manageable arrays. This doesn't fully support all of the screen_loc spec, but it's enough for what I'm doing. (f54865f)

Uses these helper procs to improve existing code (6273b93)

Fixes an issue with tooltip code itself. If you tried to hold down a mouse button while dragging onto a tooltip enabled object, it would silently fail. The js made assumptions about the order args came in, which broke when buttons were held down (e0e42f6)

Adds a signal linked to /client/Click(). Surprised we didn't have this before honestly (c491a4a)

Makes /client/MouseDrag() return parent. If we don't do this, any overrides of MouseDrag will never actually be called (2190b2a)
Refactors how action buttons work under the hood (53ccce2)
Basically, rather then generating one button per action, we generate one button per viewer

Starts to change button behavior, more cleanup

Changes the mouse cursor when an action button is dragged. Hopefully
this makes moving things feel less like an accident, and makes you doing
it more clear

Removes the moved and locked vars. This will be more relevant later, but
for now:

Moved exists as a sort of budget "We've been dragged" variable. We can
handle this more cleanly, and the movable type doesn't care about it

Locked is a very old variable that is also not something that the
movable type "owns". It's more an action button thing that's been moved
down.
It exists so an action can be locked in place, and in that locking, be
treated as a "saved location"
(21e20fc)

Because I've nuked move, we don't need to directly set our button's
position. We can use the default_button_position var instead. This is
quite handy.

Please ignore position_action, I will explain that later
(83e265e)

Removes the buttons locked pref

It was another obscure part of action buttons, basically do buttons
start "locked" or not. See previous discussion of locked
(b58b1bd)

Major rework starts here

Alright. Sorry for this, this is where me not commiting regularly starts
to suck. I'll do my best though.

Rather then figuring out an action button's position via a combination
of the moved and ordered vars, we use a separate location var to store
one of a few defines. This makes life later much easier.

Adds tooltip support for dragging action buttons. The way the tooltip
just froze in place when dragging really bugged me, and lead to some
nasty visual artifacts.
This is a bit messy because the drag procs are horrible, but it's
workable

Dropping a button on another button will no longer swap their positions
Behavior instead depends on the target button.

If it's a part of a group (A concept I will explain later) the dragged
button is simply inserted before it in the group's list.

If it's floating on the general hud, we instead position the dragged
button to its right. There's extra logic here to ensure buttons will
never overflow the screen, but I'll get into that later.

Alright. That's most of the refactoring. Time for the larger behavior
changes.

Adds a button palette. This is a separate dropdown that renders
underneath buttons.

image

The idea is to allow for a conceptual separation between "important"
buttons and the ones that end up cluttering the screen.

You can click on the dropdown to open it, then any later clicks that
don't involve actions in some way will autoclose it.

My goal is to come up with an alternative for the action button that
just acted as a way to hide all buttons on screen. Not convinced it saw
much use.

As a side effect of removing that, I've moved its tooltip stuff to the
palette. I've properly formatted it, so hopefully it's easier to read
then the jumble that we used to have.

(You can alt click the palette button to reset all button positions)

Oh and the palette can scroll, since as you'll see later it has a
limited size.
image

Moving on from that, I've added what amounts to action landing buttons.
These allow buttons to rejoin groups, or be positioned at the end of a
line of buttons.
image

They've got a 32x32 hitbox, and only show up when dragging. Hopefully
this makes the system more clear just by dragging an action.

Oh and I've changed how button position updating works. The old system
of calling update_action_buttons on mob every time an action button
changes position is gone, mostly because I've setup more robust
grouping. Will discuss when I get to huds

(0d1e93f)
Adds the backbone behind action button position changes (94133bd)

Moves hud defines to the global folder, safer this way (7260117)

Adds color changing to the palette button, giving some heads up for buttons being inserted into the palette automatically
image
image
Ensures a landing button is always shown, even if it needs to break the
max row rule
Makes palettes auto contract if they have no buttons inside them
Prevents palettes from being opened if they have no buttons inside them
(f9417f3)
How it looks
2022-02-26.02-30-10.mp4
Why It's Good For The Game

Players have more control over the clutter on their screen.
Buttons are available, but not in the way,
Since any player move of a button saves it, any lack of clarity in the way buttons work will be forced out by buttons not just resetting when a new game starts.
We don't overlap any existing screen elements, unless the upper button list gets really long.
The code is much less crummy (I think, may have made it worse it's hard for me to judge my own work)

If it ends up not being as usable as I'd like, I'll rip out the existing changes and just implement the qol and backend stuff. I think it's worth doing though.
Changelog

cl
add: Expanded heavily on action buttons
add: Adds an action button dropdown that sits just under the normal list in the top left. You can drag new buttons onto it to insert them. Click on it to show its contents, do what you want to do, then click again anywhere to contract it. Alt click it to reset all button positions
add: Action buttons will now remember their position between rounds. So if you really like your flashlight right next to your player for some reason, we support that now
add: When you start to drag an action button, docking ports will appear in places that it can be inserted into. (Outside of just floating somewhere on your screen of course)
del: Removed action button locking, and the associated preference. I'm reasonably sure literally none uses this, but if you do hit me up
qol: Dragging an action button will now give you an outline of its size around your cursor
fix: You can no longer cause the screen to expand by putting an action button on the edge of widescreen, and then resizing to standard.
refactor: Refactors action and button code significantly. lots of little things.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65180 [ci skip]

unit0016 committed:
Buffs The Ecto-Sniffer Again (#65186)


About The Pull Request

Not even a louder sound got roboticists to pay attention to this machine, so I've done it.
This adds a a 'say()' to the ecto sniffer, along with a handful of silly (and spooky) phrases for it to say.
Why It's Good For The Game

Roboticists please just make cyborgs, I'm begging you. It's the whole point of your job. I don't want to make this even more absurd.
Changelog

cl
accessibility: Ectoscopic Sniffers now shout a handful of silly phrases to remind robotics to not ignore them, and to accommodate deaf players.
/cl

Tastyfish committed:
Moves help menu to end, fixes Hotkeys Help (#65295)


About The Pull Request

    Moves the Help menu to the end of the menu bar, like every app that has ever had a Help menu.

image

    Makes Hotkeys Help actually do something, given the verb that menu item calls was removed 3 years ago.
        It now show all of your hotkeys, as set in preferences, in a TGUI table. They're in Binding sort order, as opposed to the preferences window, which may be of use to people.

image
Why It's Good For The Game

halp how do i standup
Changelog

cl
add: Re-added the Hotkeys-Help verb, and linked the Hotkeys menu item to it.
qol: Move Help menu to the end of the menu bar.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65295 [ci skip]

TiviPlus committed:
Humans no longer "enter" mechs twice (#65819)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

tgstation-server committed:
Automatic changelog generation for PR #65819 [ci skip]

TiviPlus committed:
abstract Mech air tank can no longer be removed (#65818)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

tgstation-server committed:
Automatic changelog generation for PR #65818 [ci skip]

vincentiusvin committed:
Ingame Atmos Reaction Guide (#65271)

About The Pull Request

https://imgur.com/a/pMMEi4i
https://imgur.com/a/xCrIcz4

Title, really.
Adds an ingame guide to atmos. Currently hooked to the atmos monitors, analyzer, and the tablet app.

Lots of reaction data not implemented yet, banking on the cleanup to get merged first, so drafting.
Done, all reactions in. Haven't double checked them though.

Code is pretty much finished, feel free to take a look. Ill probably retidy them while adding stuffs so no rush.

Might add a reaction handbook obj later Implemented in analyzer.

Dotted tooltip idea shamelessly stolen from preferences.

Lots of the diffs are from breaking the sensor file up, dont worry about it.
Why It's Good For The Game

Less need to open the wiki in another page I guess.
Changelog

cl
add: Added an ic atmos reaction guide. Available in your atmos control consoles/monitors, ntosatmos app, and analyzer.
code: Some changes to how gas canister descriptions are generated.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65271 [ci skip]

ivanmixo committed:
Give space dragon ability to make it's sprite small (#65311)


About The Pull Request

Allows space dragon to replace it's sprite with a carp sprite like the xeno queen can turn it's sprite into a drone.

This means only the dragon sees itself as small. Others still see it as giant.
Why It's Good For The Game

The space dragon sprite is pretty big, so you can't attack stuff directly north of you. This solves that.
Changelog

cl
qol: space dragon can now make it's sprite smaller
/cl

tgstation-server committed:
Automatic changelog generation for PR #65311 [ci skip]

san7890 committed:
Fixes commented out line in IceBoxStation.dmm (!!!) (#65837)

Fixes commented out line in IceBoxStation.dmm

Hey there,

So the library PR used some peculiar method (very jank, very sad!) to replace the paths, and this managed to creep its way in:

This would cause/throw weird issues like the following in StrongDMM:

Updating one of the effected tiles (which showed up like this):

Would result in the lower turf becoming proper Space (very bad for IceBox!).


God I'm a moron aren't I

Co-authored-by: LemonInTheDark <58055496+LemonInTheDark@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #65837 [ci skip]

JohnFulpWillard committed:
re-adds the assistant traitor toolbox (#65703)

tgstation-server committed:
Automatic changelog generation for PR #65703 [ci skip]

Commit:
Adds pack checks to departmental order console (#65804)

Department console could "buy" emagged/special/unbuyable crates prior.

tgstation-server committed:
Automatic changelog generation for PR #65804 [ci skip]

Iamgoofball committed:
You can now see plant health with the plant analyzer. (#65775)

tgstation-server committed:
Automatic changelog generation for PR #65775 [ci skip]

robbertapir committed:
Moves the deep fried food holder to it's own file (#65734)

Moves the deep fried food holder from bread.dm to deepfried.dm.
It's neither bread nor bread-related. There is no reason for it to be in bread. No reasonable person would reasonably expect it to be in bread.

Timberpoes committed:
Feex (#65612)

Splits reagent log string creation to a standard proc and a bespoke stupid proc that takes an external list instead of using its own reagents_list.

tgstation-server committed:
Automatic changelog generation for PR #65612 [ci skip]

Spookuni committed:
Turns out brain damage is harmful who knew (#65790)

tgstation-server committed:
Automatic changelog generation for PR #65790 [ci skip]

PetMudstone committed:
Consistent names and spelling. (#65736)

tgstation-server committed:
Automatic changelog generation for PR #65736 [ci skip]

SuperNovaa41 committed:
Removes some unused mob verbs + client vars (#65732)

* Removes an unused var from /client
`inprefs` isn't used anywhere, i checked to when it was added (~5 years
ago) and it wasn't even really used then. I think it used to be used to
track topic calls / slowdown some href issues but it isn't any more.
Byebye!!

* Removes some unused mob verbs + client var
`canface` and `east/west/north/southface` used to be used to set
direction but these are no longer used and are thus taking up
unnecessary space

Ghommie committed:
Changes the values of the pen colour variable from color name to hexadecimal (#65553)


About The Pull Request

See the title. Hexadecimal values are better than color names for programming since they can be sanitized and are more easily converted to RGB(A) or HSV. Color names aren't really meant to be used in some functions and methods about colors.

TL;DR: color names don't work with the Color common package in the TGUI folder.
Why It's Good For The Game

This will remove the need of a hacky bandaid from /obj/item/canvas/proc/get_paint_tool_color that only covers about half the pen colors anyway.
Changelog

cl
fix: Fixed certain pens breaking painting on a canvas because of their peculiar colors.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65553 [ci skip]

TiviPlus committed:
Better mech ammo names and new ammo defines (#65817)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

tgstation-server committed:
Automatic changelog generation for PR #65817 [ci skip]

TiviPlus committed:
COMSIG_ATOM_CREATED -> COMSIG_ATOM_INITIALIZED_ON (#65809)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

TiviPlus committed:
Fix mech category overrides (#65810)

Co-authored-by: TiviPlus <572233640+TiviPlus@users.noreply.com>

tgstation-server committed:
Automatic changelog generation for PR #65810 [ci skip]

Arkatos1 committed:
Fixes several incorrect spider menu descriptions and automatizes the process a little bit (#65806)

tgstation-server committed:
Automatic changelog generation for PR #65806 [ci skip]

capsaicinz committed:
revenant harvest now logs (for real this time!) (#65715)

tgstation-server committed:
Automatic changelog generation for PR #65715 [ci skip]

unit0016 committed:
Fixes That /One/ Tramstation Bin. You know the one. (#65763)

tgstation-server committed:
Automatic changelog generation for PR #65763 [ci skip]

Wallemations committed:
Lowers the Kheiral Cuff price to be more reasonable with their changed function (#65697)

tgstation-server committed:
Automatic changelog generation for PR #65697 [ci skip]

san7890 committed:
MetaStation Xenobiology Satellite Tweaks (#65772)

tgstation-server committed:
Automatic changelog generation for PR #65772 [ci skip]

san7890 committed:
Removes Floating Poster on MetaStation's Abandoned Bar (#65797)

tgstation-server committed:
Automatic changelog generation for PR #65797 [ci skip]

Ghilker committed:
Splits supermatter.dm into multiple files (#65805)

Kylerace committed:
fixes move to nullspace runtime in mob/abstract_move() (#65786)

actions-user committed:
Automatic changelog compile [ci skip]

Ebin-Halcyon committed:
Maid costumes, the respritening (#65844)


About The Pull Request

Haha gotchu, april fools

haha gotchu again, it's an actual resprite and additions to the maid costume with a headband, arm covers or whatever you call them, and a neck cover.

Spin arounds of the sprites.
Content
Content2
Why It's Good For The Game

TG is on a crusade to delete old sprites, or whatever. Someone has to do this at some point.
Changelog

cl
imageadd: New and updated maid costume sprites
/cl

tgstation-server committed:
Automatic changelog generation for PR #65844 [ci skip]

Ryll-Ryll committed:
Adds linking tickets in asay, referencing mob datums in asay, improves asay ref code (#65634)


About The Pull Request

Continuing on the work started in #65154 and 61712, this PR expands what asay linking can be used for and improves the code for it. You can now link tickets in asay by saying #[ticket ID] like #1. This shows the ticket's status and who filed the ahelp. Might remove the ticket filer if it's not really useful.

This also fixes an oversight of mine with datum linking, with how mobs and certain other datums (armor datums, landmarks) don't follow the memory address format for their tag values. Datum linking now uses a @[tag] format, so you can link to a mob by saying @mob_702 and such.

dreamseeker_2022-03-22_21-51-41.png
Why It's Good For The Game

More robust admin tools
Changelog

cl Ryll/Shaps
admin: You can now link tickets in asay with #[ticket ID], like #1 and #5.
admin: asay datum linking now functions by saying '@[datum's tag value]`, whether it's a normal datum (@0x2001169) or a mob (@mob_702) or whatever else.

Discord_2022-03-16_23-38-09.png
(old pic)

san7890 committed:
Adds a new critter to TramStation (#65830)


About The Pull Request

Man, look at how nice TramStation is.

image

You can really tell. It's been missing something as of late, though. Hey, what do you think is off there? That little white speck? Let's zoom in a bit.
Why It's Good For The Game

image

Ah, splendid. Tramstation really was missing this fella.
Spoiler on Location

I like things being a secret, but I understand that it's probably a good part of the review process to have people easily see what I'm referring to. I add a new lizard, Bold-and-Brash (clever, eh?) and add them near disposals to a map that had no lizard plushie (Tram). I think it's quite clever, and a nice surprise for anyone that doesn't expect it.

image
Changelog

cl
add: A new plushie has appeared on TramStation. Seek it out if you dare.
/cl

tgstation-server committed:
Automatic changelog generation for PR #65830 [ci skip]

timothymtorres committed:
Fix shutters contextual screentips (#65750)

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #65750 [ci skip]

Ghilker commented:
ah fuck

Ghilker commented:
i uh sorry people

Ghilker committed:
merge master

rdragan commented:
try invoking a sink

Ghilker commented:
I think I'll close this one and remake it cause this is messed up badly

LemonInTheDark commented:
chad
